### PR TITLE
lido liquidity updates

### PR DESCRIPTION
**Pull request by ppclunghe**
Adding new wstETH pools to Lido liquidity spell, update for Uniswap wstETH pools spell on Base

### CODE REVIEW

File: models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    schema='lido_liquidity_base',
+    alias = 'aerodrome_cl',     
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool', 'time'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "lido_liquidity",
+                                \'["pipistrella"]\') }}'
```
Comment (Hosuke) [inline models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql:11]:
```suggestion
    post_hook='{{ expose_spells(blockchains = \'["base"]\',
                                spell_type = "project",
                                spell_name = "lido_liquidity",
                                contributors = \'["pipistrella"]\') }}'
```

File: models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+      cr.token1,
+      SUM(CAST(amount0 AS DOUBLE)) AS amount0,
+      SUM(CAST(amount1 AS DOUBLE)) AS amount1
+ from {{source('aerodrome_base','CLPool_evt_Swap')}} s
+ left join {{source('aerodrome_base','CLFactory_evt_PoolCreated')}} cr on s.contract_address = cr.pool
+ 
+ {% if not is_incremental() %}
+  WHERE DATE_TRUNC('day', s.evt_block_time) >= DATE '{{ project_start_date }}'
+ {% else %}
+  WHERE DATE_TRUNC('day', s.evt_block_time) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
```
Comment (Hosuke) [inline models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql:163]: Maybe we can use incremental predicate for incremental filters:
```suggestion
  WHERE {{ incremental_predicate('s.evt_block_time') }}
```

File: models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      LEFT JOIN tokens_prices_daily AS p0 ON l.time = p0.time
+      AND l.token0 = p0.token
+      LEFT JOIN tokens_prices_daily AS p1 ON l.time = p1.time
+      AND l.token1 = p1.token
+      LEFT JOIN trading_volume AS tv ON l.time = tv.time
+      AND l.pool = tv.pool
+ )
+
+ 
+select  CONCAT(CONCAT(CONCAT(CONCAT(CONCAT(blockchain,CONCAT(' ', project)) ,' '), coalesce(paired_token_symbol,'unknown')),':') , main_token_symbol, ' ', pool_type, ' ', format('%,.3f',round(coalesce(fee,0),4))) as pool_name,
```
Comment (Hosuke) [inline models/lido/liquidity/base/lido_liquidity_base_aerodrome_cl_pools.sql:304]: I am not sure `concat` can be replaced by `||` here.

File: models/lido/liquidity/scroll/lido_liquidity_scroll_syncswap_pools.sql [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    schema='lido_liquidity_scroll',
+    alias = 'syncswap_pools',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool', 'time'],
+    post_hook='{{ expose_spells(\'["scroll"]\',
+                                "project",
+                                "lido_liquidity",
+                                \'["pipistrella"]\') }}'
```
Comment (Hosuke) [inline models/lido/liquidity/scroll/lido_liquidity_scroll_syncswap_pools.sql:11]: Same as above post_hook.

File: models/lido/liquidity/scroll/lido_liquidity_scroll_zebra_pools.sql [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    schema='lido_liquidity_scroll',
+    alias = 'zebra_pools',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool', 'time'],
+    post_hook='{{ expose_spells(\'["scroll"]\',
+                                "project",
+                                "lido_liquidity",
+                                \'["pipistrella"]\') }}'
```
Comment (Hosuke) [inline models/lido/liquidity/scroll/lido_liquidity_scroll_zebra_pools.sql:11]: Same as above.

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @ppclunghe,

we can apply the new format for post_hooks in this PR.

Also I think we may use `||` to replace `concat` to shorten the long line.

File: models/lido/liquidity/scroll/lido_liquidity_scroll_zebra_pools.sql (hunk 5 above)
Comment (ppclunghe) [reply models/lido/liquidity/scroll/lido_liquidity_scroll_zebra_pools.sql]: Thanks a lot for your review, @Hosuke 
I have updated code according to your comments

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ppclunghe 

[Sample lido.liquidity]()

**Comment by ppclunghe** [comment]
I have added  a all fix for SyncSwap wstETH pools on zksync, sorry for it, all comments from @Hosuke were taken into account

File: models/lido/liquidity/zksync/lido_liquidity_zksync_syncswap_v2_pools.sql [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+    schema='lido_liquidity_zksync',
+    alias = 'syncswap_v2_pools',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool', 'time'],
+    post_hook='{{ expose_spells(\'["zksync"]\',
+                                "project",
+                                "lido_liquidity",
+                                \'["pipistrella"]\') }}'
```
Comment (Hosuke) [inline models/lido/liquidity/zksync/lido_liquidity_zksync_syncswap_v2_pools.sql:11]:
```suggestion
    post_hook='{{ expose_spells(blockchains = \'["zksync"]\',
                                spell_type = "project",
                                spell_name = "lido_liquidity",
                                contributors = \'["pipistrella"]\') }}'
```
```suggestion
    post_hook='{{ expose_spells(\'["zksync"]\',
                                "project",
                                "lido_liquidity",
                                \'["pipistrella"]\') }}'
```

File: models/lido/liquidity/base/lido_liquidity_base_uniswap_v3_pools.sql [hunk 7]
```diff
@@ -63,7 +63,7 @@ with
     {% if not is_incremental() %}
     WHERE DATE_TRUNC('day', p.minute) >= DATE '{{ project_start_date }}'
     {% else %}
-    WHERE DATE_TRUNC('day', p.minute) >= DATE_TRUNC('day', NOW() - INTERVAL '1' day)
+    WHERE {{ incremental_predicate('p.minute') }}
```
Comment (jeff-dude) [inline models/lido/liquidity/base/lido_liquidity_base_uniswap_v3_pools.sql:66]: similar to the changes made here to use incremental predicate on source reads, we also leverage them for filtering the target data. this allows the generated merge statement to filter both to same time period and speed up processing. i won't consider a blocker for now, but it'd be nice to see a follow up PR soon that implements on all lido models. looks like this:
https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/arbitrum/platforms/airswap_arbitrum_base_trades.sql#L9

be sure the column name passed in matches target spell column name, sometimes it changes from source to target

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**